master.js

```diff
@@ -1007,15 +962,74 @@ async function getPlugins(){
 		let plugin = plugins[i];
 		if(plugin.main.onLoadFinish && typeof plugin.main.onLoadFinish == "function") await plugin.main.onLoadFinish({plugins});
 	}
-	console.log("All plugins loaded in "+(Date.now() - startPluginLoad)+"ms");
 	return plugins;
 }
 
+async function startServer() {
+	// Set the process title, shows up as the title of the CMD window on windows
+	// and as the process name in ps/top on linux.
+	process.title = "clusterioMaster";
+
+	// add better stack traces on promise rejection
+	process.on('unhandledRejection', r => console.log(r));
 
-masterPlugins = await getPlugins();
+	/** Sync */
+	function randomStringAsBase64Url(size) {
+	  return base64url(crypto.randomBytes(size));
+	}
+
+	if (!fs.existsSync("secret-api-token.txt")) {
+		config.masterAuthSecret = randomStringAsBase64Url(256);
+		fs.writeFileSync("config.json",JSON.stringify(config, null, 4));
+		fs.writeFileSync("secret-api-token.txt", jwt.sign({ id: "api" }, config.masterAuthSecret, {
+			expiresIn: 86400*365 // expires in 1 year
+		}));
+		console.log("Generated new master authentication private key!");
+		process.exit(0);
+	}
+	// write an auth token to file
+	fs.writeFileSync("secret-api-token.txt", jwt.sign({ id: "api" }, config.masterAuthSecret, {
+		expiresIn: 86400*365 // expires in 1 year
+	}));
+	setInterval(()=>{
+		fs.writeFileSync("database/items.json", JSON.stringify(db.items));
+	},config.autosaveInterval || 60000);
+	process.on('SIGINT', shutdown); // ctrl + c
+	process.on('SIGHUP', shutdown); // terminal closed
+
+	// Make sure we're actually going to listen on a port
+	let httpPort = args.masterPort || config.masterPort;
+	let httpsPort = args.sslPort || config.sslPort;
+	if (!httpPort && !httpsPort) {
+		console.error("Error: at least one of httpPort and sslPort must be configured");
+		process.exit(1);
+	}
 
+	// Load plugins
+	await pluginManagement();
 
-io.attach(server);
+	// Only start listening for connections after all plugins have loaded
+	if (httpPort) {
+		let httpServer = require("http").Server(app);
+		httpServer.listen(httpPort);
+		console.log("Listening for HTTP on port %s...", httpServer.address().port);
+	}
+
+	if (httpsPort) {
+		var certificate = fs.readFileSync(path.resolve(config.databaseDirectory, 'certificates/cert.crt'));
```

Comment:
This can also go into config and use, just have a default value in it.

Reply:
This already has a config option called sslCert, and I already have a fix this.  See the "Use sslCert and sslPrivKey config entries" commit of my config-rewrite branch.